### DISCUSSION TITLE
showing lineage data from external file in spline UI

### DISCUSSION THREAD

**Discussion by KumarMutt** [AbsaOSS/spline-getting-started#51] [General]
```shell
spark-submit \
   --packages za.co.absa.spline.agent.spark:spark-3.3-spline-agent-bundle_2.12:2.0.0 \
   --conf spark.sql.queryExecutionListeners=za.co.absa.spline.harvester.listener.SplineQueryExecutionListener \
   --conf spark.spline.lineageDispatcher=hdfs \
   --conf spark.spline.producer.url=http://localhost:8080/producer   my_spark_job.py
```

   I am using above code to store lineage information into external file now i want to see this data in spline UI by reading external file instead of directly getting in http request can custom lineageDispatcher useful here?

**Comment by wajda** [comment]
No, currently there is no end-user-friendly way (like a cli-tool or a UI) to import lineage from external files. You need to call `Producer REST API` to ingest lineage data. Also you may adopt our test script for this purpose.
https://github.com/AbsaOSS/spline/blob/develop/test-data-generator/send-plans-and-events.sh
